src/Services/Catalog/Catalog.Application/UseCases/Categories/UpdateCategory/UpdateCategoryInteractor.cs

```diff
@@ -0,0 +1,33 @@
+﻿using AutoMapper;
+using Catalog.Domain.Abstractions;
+using MediatR;
+
+namespace Catalog.Application.UseCases.Categories.UpdateCategory;
+
+public class UpdateCategoryInteractor : IRequestHandler<UpdateCategoryUseCase, string>
+{
+    private readonly ICategoryRepository _categoryRepository;
+    private readonly IMapper _mapper;
+
+    public UpdateCategoryInteractor(ICategoryRepository categoryRepository, IMapper mapper)
+    {
+        _categoryRepository = categoryRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<string> Handle(UpdateCategoryUseCase request, CancellationToken cancellationToken)
+    {
+        Domain.Entities.Category? category = await _categoryRepository.GetByIdAsync(request.CategoryId, cancellationToken);
+
+        if (category is null)
```

Comment:
use same style of writing your code. use Domain.Entities.Category category = await _categoryRepository.GetByIdAsync(request.CategoryId, cancellationToken)
            ?? throw new KeyNotFoundException("category with specified id wasn't found"); or use another style everywhere